### PR TITLE
feat: copilot plugin debugger

### PR DESCRIPTION
**Pull request by jayzhang**
feature link: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/29080474

feature flag: `TEAMSFX_API_PLUGIN_DEBUG=true`  （default value is false）

image

### CODE REVIEW

**Review by huimiu — DISMISSED** [review]

**Review by XiaofuHuang — DISMISSED** [review]

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts [hunk 1]
```diff
@@ -453,6 +455,9 @@ async function onDidEndTaskProcessHandler(event: vscode.TaskProcessEndEvent): Pr
 }
 
 async function onDidStartDebugSessionHandler(event: vscode.DebugSession): Promise<void> {
+  if (isM365CopilotChatDebugSession(event)) {
+    void cdpClient.start();
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/teamsfxTaskHandler.ts]: Does this correctly handle corner case like start twice?
One possible reason is that user stops then restarts debugging in a short period.

File: packages/vscode-extension/src/debug/common/debugConstants.ts [hunk 2]
```diff
@@ -254,6 +254,16 @@ export const launchingTeamsClientDisplayMessages = Object.freeze({
     "The app supports hot reloading. If you have any code changes in the project, the app will be reloaded.",
 });
 
+export const connectToExistingBrowserDebugSessionForCopilot = Object.freeze({
+  title: "Connecting to Existing Browser Debug Session for Copilot.",
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/debug/common/debugConstants.ts:268]: Will you localize these strings in later PR?

**Review by a1exwang — DISMISSED** [review]

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts (hunk 1 above)
Comment (jayzhang) [reply packages/vscode-extension/src/debug/teamsfxTaskHandler.ts]: Yes, I will fix it.

**Review by tecton — DISMISSED** [review]

File: packages/vscode-extension/src/debug/common/debugConstants.ts (hunk 2 above)
Comment (jayzhang) [reply packages/vscode-extension/src/debug/common/debugConstants.ts:268]: Not for this version

**Review by HuihuiWu-Microsoft — DISMISSED** [review]

**Review by swatDong — APPROVED** [review]

**Review by huimiu — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Pull request merged**